runtimes/zephyr/src/main.c

```diff
@@ -318,3 +319,8 @@ void main(void) {
   posix_exit(0);
 #endif
 }
+
+tvm_crt_error_t TVMPlatformGenerateRandom(uint8_t* buffer, size_t num_bytes) {
+  sys_rand_get(buffer, num_bytes);
```

Comment:
which impl of random are you using? I had used the timer one and it's buggy for num_bytes<4: https://github.com/zephyrproject-rtos/zephyr/blob/d4347b4015c9a5992877b3086508b134e5fa87b3/subsys/random/rand32_timer.c#L73

Reply:
Yeah, no idea -- it's whatever gets baked in by default. I am using CONFIG_ENTROPY_GENERATOR. Is there a way I can tell what is getting compiled in?

Reply:
can you check this `.config` file in the workspace build directory:

`workspace/builds/2021-01-15T18-26-35/build/runtime/zephyr/.config`

(replace the datestamp with one that makes sense to you)

Reply:
This is what I see:
```
#
# Random Number Generators
#
CONFIG_ENTROPY_DEVICE_RANDOM_GENERATOR=y
# CONFIG_XOROSHIRO_RANDOM_GENERATOR is not set
CONFIG_CSPRING_ENABLED=y
CONFIG_HARDWARE_DEVICE_CS_GENERATOR=y
# end of Random Number Generators
```

Not sure which one is the one you had problems with.

Reply:
cool, it was not that, it was `CONFIG_TIMER_RANDOM_GENERATOR`. I think your board does not enable that impl, but since the STM32F746xx needs it, it'd be helpful to change the common code so it's compatible with both. the impl from the TVM repo (https://github.com/apache/tvm/blob/main/tests/micro/qemu/zephyr-runtime/src/main.c#L165) has been tested on STM32F746xx. Do you mind trying that impl here? it should be compatible

Reply:
Done! PTAL